src/main/scala/BallCore/Rest/RestManager.scala

```diff
@@ -0,0 +1,73 @@
+package BallCore.Rest
+
+import BallCore.Storage.{Migration, SQLManager}
+import cats.effect.IO
+import skunk.Session
+import skunk.codec.all.*
+import skunk.implicits.*
+
+import java.util as ju
+import java.util.UUID
+
+val DEFAULT_REST = 1.0
+
+/** Manages rest related tasks and utilities.
+  */
+class RestManager()(using sql: SQLManager):
+    sql.applyMigration(
+        Migration(
+            "Initial Rest Manager",
+            List(
+                sql"""
+                    CREATE TABLE RestValues (
+                        PlayerID UUID NOT NULL,
+                        Rest DOUBLE PRECISION NOT NULL,
+                        UNIQUE(PlayerID)
+                    );
+                    """.command
+            ),
+            List(
+                sql"""DROP TABLE RestValues;""".command
+            ),
+        )
+    )
+
+    /** Stores a player's rest value in the database.
+      *
+      * @param playerId
+      *   The player's Minecraft UUID.
+      * @param rest
+      *   The player's rest value.
+      * @param session
+      *   A connection to the database.
+      */
+    def save(playerId: UUID, rest: Double)(using
```

Comment:
i would have a caching layer to avoid roundtripping the database every time a rest value is needed; a `TrieMap` from UUID to rest value should suffice